### PR TITLE
Added -a/--all to `mafia quick` and `mafia watch`

### PR DESCRIPTION
**Pull request by jacobstanley**
/cc @charleso @nhibberd

This adds support for loading the source of all libraries when using quick/watch. Seems like `ghci` is smart enough to only load the things it needs, pretty cool.

Not sure if `src`, `gen` and `dist/build/autogen` are the right sub-directories to be including or not? Seems to work ok :)

### CODE REVIEW

**Comment by nhibberd** [comment]
woah

File: main/mafia.hs [hunk 1]
```diff
@@ -257,6 +274,19 @@ ghciArgs extraIncludes path = do
             <> (fmap ("-package-db=" <>) databases)
             <> [ path ]
 
+reifyInclude :: GhciInclude -> EitherT MafiaViolation IO [Directory]
+reifyInclude = \case
+  Directory dir -> return [dir]
+  AllLibraries  -> do
+    absDirs <- Set.toList <$> getSubmoduleSources
+    relDirs <- mapM tryMakeRelativeToCurrent absDirs
+    return [ dir </> sub | dir <- relDirs
```
Comment (nhibberd) [inline main/mafia.hs:283]: Not sure about this bit

**Comment by charleso** [comment]
Not sure what the downside/implications are, but happy to be a guinea :pig: to try this out :+1:

File: main/mafia.hs (hunk 1 above)
Comment (jacobstanley) [inline main/mafia.hs:283]: Would this be better?

``` haskell
    return $ do
      dir <- relDirs
      sub <- ["src", "test", "gen", "dist/build/autogen"]
      return (dir </> sub)
```

Comment (jacobstanley) [inline main/mafia.hs:283]: Or maybe this?

``` haskell
    return $ (</>) <$> relDirs <*> ["src", "test", "gen", "dist/build/autogen"]
```

Comment (tranma) [inline main/mafia.hs:283]: :fearful:

Comment (jacobstanley) [inline main/mafia.hs:283]: Ok, talked to @nhibberd IRL, sticking with the list comprehension

**Pull request merged**